### ISSUE TITLE
Save Metadata in PACertificate

### ISSUE THREAD

**Issue by dampfklon** [rmbolger/Posh-ACME#441]
Can you somehow save metadata to a PACertificate?

The feature would be to save some data when creating a new Certificate and retrieving this metadata on renewal.

A use case would be to save some data on how the certificate gets deployed on creation to use it later on renewal.

I would image the same principle like the PluginArgs which get saved along the other certificate data.

**Comment by dampfklon** [comment]
Thanks for your feedback. 
We have a central server which uses Posh-ACME to generate multiple certificates for different systems. I have scripts which deploy to those systems. My use case would be to save which script is needed to deploy the certificate on initial creation.

Your suggestion to save the data along in the order's folder and simply load it after renewal would be a feasible solution. Especially if such a feature in the base module would raise security concerns.

**Comment by rmbolger** [comment]
Yeah, that should work just fine for the time being. It's mildly abusing module implementation details that could possibly change. But I have no current plans to change the profile folder structure and if it ever does change, it would only be in a major version update and the `.Folder` parameter on the order would still reflect the order's folder path wherever that might be.
